### PR TITLE
Improve stored config for inference

### PR DESCRIPTION
**Pull request by JingyaHuang**
The way for storing neuron-related configuration done in #117 is better than what I put in place, this PR is inspired to better organize neuron config.

### CODE REVIEW

File: optimum/neuron/utils/argument_utils.py [hunk 1]
```diff
@@ -143,29 +143,32 @@ def store_compilation_config(
...
-    update_func("neuron_compiler", neuron_compiler)
-    update_func("neuron_compiler_version", neuron_compiler_version)
+    neuron_configs["compiler_type"] = compiler_type
+    neuron_configs["compiler_version"] = compiler_version
 
     # Add input shapes during compilation to the config
     for axe, shape in input_shapes.items():
-        axe = f"neuron_{axe}"
-        update_func(axe, shape)
+        axe = f"static_{axe}"
```
Comment (dacorvo) [inline optimum/neuron/utils/argument_utils.py:162]: This is really a detail, but `axe` is french, isn't it ? what you really mean is `axis`, not 🪓 right ?

File: optimum/neuron/utils/argument_utils.py [hunk 2]
```diff
@@ -143,29 +143,32 @@ def store_compilation_config(
...
-    neuron_compiler_version: str,
+    compiler_type: str,
+    compiler_version: str,
     **kwargs,
 ):
     if isinstance(config, OrderedDict):
         update_func = config.__setitem__
     else:
         update_func = config.__setattr__
+    neuron_configs = {}
```
Comment (dacorvo) [inline optimum/neuron/utils/argument_utils.py:154]: You seem to call it `neuron_config` (singular) in other places.

**Review by dacorvo — COMMENTED** [review]
LGTM. Just two really minor comments.

File: optimum/neuron/utils/argument_utils.py (hunk 1 above)
Comment (JingyaHuang) [reply optimum/neuron/utils/argument_utils.py:162]: That's true... Will change it!

**Review by dacorvo — APPROVED** [review]

File: optimum/neuron/utils/argument_utils.py (hunk 2 above)
Comment (JingyaHuang) [reply optimum/neuron/utils/argument_utils.py:154]: As the `neuron_configs` here is not the same thing as `neuron_config`(instance of `NeuronConfig`), I should probably not name them the same. I might just go with `config_args`.

**Pull request merged**